src/saga/adaptors/slurm/slurm_job.py

```diff
@@ -479,7 +479,7 @@ def _job_run (self, jd) :
             # we start N independent processes
             mpi_cmd = ''
 
-            if self._version >= StrictVersion('17.11.5'):
+            if self._version in [StrictVersion('17.11.5'),StrictVersion('18.08.3')]:
```

Comment:
Why did the `>=` did not accomplish the same?  What happens if Stampede is moving to `18.08.4`?  If `StrictVersion` is not working as intended,  we should not use it at all -- it will then only obfuscate the code...

Reply:
It did not because we haven't found a way to recognize the correct number of cores per node. 

Specifically, on [slurm_job.py#L388](https://github.com/radical-cybertools/saga-python/blob/fix/slurm_versioning/src/saga/adaptors/slurm/slurm_job.py#L388) we discover the number of cores per node and grab the first number always. Although that is not the case of all queues as #692 discusses. That is not the case for the use cases we support on Stampede2. The first value is the correct one.

I agree that `StrictVersion` is not needed right now, but I think it will be useful when we are able to get the correct information for `cores_per_node`

Reply:
I'll remove it. I am not sure that going with `hostnames` will solve our problem. I remember at some point we said that we may want this kind of input to be configurable. It may make more sense to have these values in a json file like:
```json
{"resourse": {"queue1" : cores,
              "queue2" : cores}
....
```

This might make our life easier, especially with queue were the nodes can be different, such the GPU queues of Bridges, where depending on the GPU type we get a different number of cores per node.

Reply:
Yes, that will be the right thing to do.  I am working on an overhaul of the SAGA config management (in `feature/json_configs`) - but until then this is what we have *sigh*.  Thanks!